sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -232,7 +232,9 @@ object GpuDeviceManager extends Logging {
       val conf = rapidsConf.getOrElse(new RapidsConf(SparkEnv.get.conf))
       val info = Cuda.memGetInfo()
 
-      val poolAllocation = computeRmmPoolSize(conf, info)
+      val poolAllocation = if (conf.isPooledMemEnabled && !"none".equalsIgnoreCase(conf.rmmPool)) {
```

Comment:
it feels that this logic belongs inside of `computeRmmPoolSize`, since that's where the exception you are seeing comes from (at least it puts it nearby).

Reply:
In line 288 we are going to say "pool size = 0 MB", it seems we should not mention the pool in this scenario (since there isn't one.)